File: CompetitionSport/src/main/java/model/Terrain.java
package model;

import java.util.List;

import javax.persistence.Column;
import javax.persistence.ElementCollection;
import javax.persistence.Embedded;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Inheritance;
import javax.persistence.InheritanceType;

@Entity
@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
public class Terrain {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer id;
	private String nom;
	@Embedded
	private Adresse adresse;
	@ElementCollection()
	@Enumerated(EnumType.STRING)
	private List<Discipline> disciplines;
	@Enumerated(EnumType.STRING)
	@Column(columnDefinition = "ENUM('Courts', 'Dojo', 'Gymnase', 'Hippodrome', 'Piscine', 'Piste', 'Skatepark', 'Stade', 'Velodrome')")
	private TypeTerrain typeTerrain;
	
	
	
	public Terrain() {
	}



	public Terrain(Integer id, String nom, Adresse adresse, List<Discipline> disciplines, TypeTerrain typeTerrain) {
		super();
		this.id = id;
		this.nom = nom;
		this.adresse = adresse;
		this.disciplines = disciplines;
		this.typeTerrain = typeTerrain;
	}



	public Integer getId() {
		return id;
	}



	public void setId(Integer id) {
		this.id = id;
	}



	public String getNom() {
		return nom;
	}



	public void setNom(String nom) {
		this.nom = nom;
	}



	public Adresse getAdresse() {
		return adresse;
	}



	public void setAdresse(Adresse adresse) {
		this.adresse = adresse;
	}



	public List<Discipline> getDisciplines() {
		return disciplines;
	}



	public void setDisciplines(List<Discipline> disciplines) {
		this.disciplines = disciplines;
	}



	public TypeTerrain getTypeTerrain() {
		return typeTerrain;
	}



	public void setTypeTerrain(TypeTerrain typeTerrain) {
		this.typeTerrain = typeTerrain;
	}



	@Override
	public String toString() {
		return "Terrain [id=" + id + ", nom=" + nom + ", adresse=" + adresse + ", disciplines=" + disciplines
				+ ", typeTerrain=" + typeTerrain + "]";
	}

	
}

File: CompetitionAngular/src/app/model/terrain.ts
import { TypeTerrain } from './type-terrain';
import { Discipline } from './discipline';
import { Adresse } from './adresse';
export class Terrain {
  public constructor(
    private _id?: number | undefined,
    private _nom?: string | undefined,
    private _adresse?: Adresse | undefined,
    private _disciplines?: Set<Discipline> | undefined,
    private _typeTerrain?: TypeTerrain | undefined
  ) {}

  public get id(): number | undefined {
    return this._id;
  }

  public set id(value: number | undefined) {
    this._id = value;
  }

  public get nom(): string | undefined {
    return this._nom;
  }

  public set nom(value: string | undefined) {
    this._nom = value;
  }

  public get adresse(): Adresse | undefined {
    return this._adresse;
  }

  public set adresse(value: Adresse | undefined) {
    this._adresse = value;
  }

  public get disciplines(): Set<Discipline> | undefined {
    return this._disciplines;
  }

  public set discipline(value: Set<Discipline> | undefined) {
    this._disciplines = value;
  }

  public get typeTerrain(): TypeTerrain | undefined {
    return this._typeTerrain;
  }

  public set typeTerrain(value: TypeTerrain | undefined) {
    this._typeTerrain = value;
  }
}
